azure-pipelines.yml

```diff
@@ -14,139 +14,48 @@ pr:
-        npm run lint
-      displayName: Run Lint
-  ##### Verify NPM and Yarn are in sync #####
-  - job: SyncPackageManagers
-    displayName: 'Verify NPM & Yarn In-Sync [Local Copy of Target Branch Must Be Up to Date]'
-    pool:
-      vmImage: 'windows-latest'
-    steps:
-    - task: UsePythonVersion@0
-      inputs:
-        versionSpec: '3.x'
-        addToPath: true
-        architecture: 'x64'
-    - task: PythonScript@0
-      continueOnError: true
-      inputs:
-        scriptSource: 'filePath'
-        scriptPath: 'dependency-verifier.py'
-        arguments: '$(System.PullRequest.TargetBranch)'
-  ##### Package and Publish #####
-  - job: Package
-    displayName: 'Package and Publish'
-    dependsOn:
-    - Build
-    - TSLint
-    condition: succeeded()
-    pool:
-      vmImage: 'windows-latest'
-    strategy:
-      matrix:
-        Runtime:
-          dir-name: 'vscode-dotnet-runtime-extension'
-          package-name: 'vscode-dotnet-runtime'
-        SDK:
-          dir-name: 'vscode-dotnet-sdk-extension'
-          package-name: 'vscode-dotnet-sdk'
-    steps:
-    - task: NodeTool@0
-      inputs:
-        versionSpec: '16.10.0'
-      displayName: 'Install Node.js'
-    - bash: |
-        if ([ $(is-sdk-release) = 'True' ] && [ $(package-name) = 'vscode-dotnet-sdk' ]) || ([ $(is-runtime-release) = 'True' ] && [ $(package-name) = 'vscode-dotnet-runtime' ]); then
-          VERSION=`node -p "require('./package.json').version"`
-        else
-          VERSION_NUM=`node -p "require('./package.json').version"`
-          VERSION="$VERSION_NUM-alpha-$(Build.BuildId)"
-        fi
-        npm version $VERSION --allow-same-version
-        echo "##vso[task.setvariable variable=version;isOutput=true]$VERSION"
-      name: GetVersion
-      displayName: 'Get Version'
-      workingDirectory: $(dir-name)
-    - bash: |
-          npm install rimraf --reg https://registry.npmjs.org/ --verbose
-          npm install vsce -g --reg https://registry.npmjs.org/ --verbose
-          vsce package -o $(package-name)-$(GetVersion.version).vsix --ignoreFile ../.vscodeignore --yarn
-      displayName: Package Artifact
-      workingDirectory: $(dir-name)
-    - task: CopyFiles@2
-      displayName: 'Copy Artifact'
-      inputs:
-        SourceFolder: '$(Build.SourcesDirectory)'
-        Contents: '**\*.vsix'
-        TargetFolder: '$(Build.ArtifactStagingDirectory)'
-        flattenFolders: true
-    - task: PublishPipelineArtifact@0
-      inputs:
-        targetPath: '$(Build.ArtifactStagingDirectory)'
-        artifactName: '$(dir-name)'
+  - ${{ each image in parameters.pools }}:
+    - template: pipeline-templates/build-test.yaml
+      parameters:
+        pool:
+          vmImage: ${{ image.vmImage }}
+          os: ${{ image.os }}
+        oes: false
+  - template: pipeline-templates/upstream-verify.yaml
+    parameters:
+      pool:
+        vmImage: windows-latest
```

Comment:
Same for this and the other 2 jobs below. Use the `NetCore-Public` pool with an image from the list, such as `windows.vs2022preview.amd64.open`.